lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -3060,3 +3060,38 @@ NormalProtocolConformance *GetImplicitConcurrentValueRequest::evaluate(
 
   return conformance;
 }
+
+namespace {
+class UsageChecker : public ASTWalker {
```

Comment:
I recommend folding this into `MiscDiagnostics.cpp`, which has a whole bunch of these structural checks.